code/game/machinery/vending_types.dm

```diff
@@ -766,12 +766,50 @@
 	premium = list(
 		/obj/item/reagent_containers/pill/mortaphenyl = 4
 	)
-	random_itemcount = 0
+	random_itemcount = FALSE
 	temperature_setting = -1
 	light_color = LIGHT_COLOR_GREEN
 	obj_flags = OBJ_FLAG_MOVES_UNSUPPORTED
 	manufacturer = "zenghu"
 
+/obj/machinery/vending/wallpharm
+	name = "\improper NanoPharm Mini"
+	desc = "A wall-mounted pharmaceuticals vending machine packed with over-the-counter bottles. For the sick salaried worker in you."
+	icon_state = "wallpharm"
+	density = FALSE
+	products = list(
+		/obj/item/storage/pill_bottle/antidexafen = 4,
+		/obj/item/storage/pill_bottle/dexalin = 4,
+		/obj/item/storage/pill_bottle/dylovene = 4,
+		/obj/item/stack/medical/bruise_pack = 5,
+		/obj/item/stack/medical/ointment = 5,
+		/obj/item/reagent_containers/hypospray/autoinjector/inaprovaline = 2,
+		/obj/item/storage/pill_bottle/vitamin = 5,
+		/obj/item/storage/pill_bottle/cetahydramine  = 4, 
+		/obj/item/storage/pill_bottle/caffeine = 3, 
+		/obj/item/storage/pill_bottle/nicotine  = 4, 
+		/obj/item/storage/pill_bottle/rmt = 2
+	)
+	prices = list(
+		/obj/item/storage/pill_bottle/antidexafen = 70,
+		/obj/item/storage/pill_bottle/dexalin = 60,
+		/obj/item/storage/pill_bottle/dylovene = 75,
+		/obj/item/storage/pill_bottle/vitamin = 55,
+		/obj/item/storage/pill_bottle/cetahydramine  = 60, 
+		/obj/item/storage/pill_bottle/caffeine = 90, 
+		/obj/item/storage/pill_bottle/nicotine  = 85, 
+		/obj/item/storage/pill_bottle/rmt = 550
+	)
+	contraband = list(
+		/obj/item/reagent_containers/pill/tox = 3,
+		/obj/item/storage/pill_bottle/perconol = 3
+	)
+	random_itemcount = FALSE
+	temperature_setting = -1
+	light_color = COLOR_GOLD
+	obj_flags = OBJ_FLAG_MOVES_UNSUPPORTED
+	manufacturer = "zenghu"
```

Comment:
```suggestion
	manufacturer = "nanotrasen"
```

Reply:
Sure! I was weary about lore having an issue with it, since that's a can of worms, but if that happens it's easy enough to change it back